test/test-waitForTransition.js

```diff
@@ -18,24 +18,43 @@ buster.testCase('waitForTransition',
 
     this.testObject.waitForTransition(target, callback);
 
+    var transitionEvents = target.one.args[0][0];
+
     // listens once
-    assert.calledWith(target.one, this.testObject._transitionEndEvents);
+    assert.calledWith(target.one, transitionEvents);
     refute.called(target.on);
   },
 
+  'Turns off target transitions events after one time attachment on the target element': function()
+  {
+    var target = {one: this.spy(), on: this.spy(), off: this.spy()}
+      , callback = this.spy()
+      ;
+
+    this.testObject.waitForTransition(target, callback);
+
+    var transitionEvents = target.one.args[0][0];
+
+    target.one.getCall(0).callArg(1);
+    // assert off is called once with transition events
+    assert.calledOnceWith(target.off, transitionEvents);
```

Comment:
You mean trigger ```one```, to check if ```off``` is called only once? coz if we repeat target.one.getCall(0).callArg(1), off will be called more than once. Sorry just wanted to be a bit more clear, may be misunderstood.